tools/common/Application.cs

```diff
@@ -209,6 +211,38 @@ public void ParseI18nAssemblies (string i18n)
 			I18n = assemblies;
 		}
 
+		public bool IsTodayExtension {
+			get {
+				return ExtensionIdentifier == "com.apple.widget-extension";
+			}
+		}
+
+		public bool IsWatchExtension {
+			get {
+				return ExtensionIdentifier == "com.apple.watchkit";
+			}
+		}
+
+		public bool IsTVExtension {
+			get {
+				return ExtensionIdentifier == "com.apple.tv-services";
+			}
+		}
+
+		public string ExtensionIdentifier {
+			get {
+				if (!IsExtension)
+					return null;
+
+				var info_plist = Path.Combine (AppDirectory, "Info.plist");
+				var plist = Driver.FromPList (info_plist);
+				var dict = plist.Get<PDictionary> ("NSExtension");
+				if (dict == null)
+					return null;
+				return dict.GetString ("NSExtensionPointIdentifier");
```

Comment:
how many times is this called ? directly or indirectly
since it's read only maybe it should be cached ?

Reply:
Good idea: https://github.com/xamarin/xamarin-macios/issues/9368